libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/ItemizedOverlay.java

```diff
@@ -40,19 +46,23 @@ public void draw(final Canvas canvas, final MapView mapView, final boolean shado
     if (shadow)
       return;
 
-    final float scale = mapView.getContext().getResources().getDisplayMetrics().density;
     final float orientation = mapView.getMapOrientation();
 
     final Projection pj = mapView.getProjection();
     for (int i = items_.size() -1; i >= 0; --i) {
       final Item item = items_.get(i);
       pj.toPixels(item.getPoint(), screenCoords);
-      onDrawItem(canvas, item, screenCoords, scale, orientation);
+      onDrawItem(canvas, item, i, screenCoords, scale(mapView), orientation);
     }
   }
 
+  float scale(final MapView mapView) {
+    return mapView.getContext().getResources().getDisplayMetrics().density;
+  }
+
   private void onDrawItem(final Canvas canvas,
                           final Item item,
+                          final int index,
```

Comment:
Please could we rename this parameter to `itemIndex`...
- here
- in the declaration of the `drawTextOnMarker` in this class
- similarly, in the declaration of the overrides `drawTextOnMarker` method in `WaymarkOverlay`